mediaapi/routing/download.go

```diff
@@ -367,10 +382,50 @@ func (r *downloadRequest) respondFromLocalFile(
 		" plugin-types application/pdf;" +
 		" style-src 'unsafe-inline';" +
 		" object-src 'self';"
-	w.Header().Set("Content-Security-Policy", contentSecurityPolicy)
 
-	if _, err := io.Copy(w, responseFile); err != nil {
-		return nil, fmt.Errorf("io.Copy: %w", err)
+	if !r.multipartResponse {
+		w.Header().Set("Content-Security-Policy", contentSecurityPolicy)
+		if _, err = io.Copy(w, responseFile); err != nil {
+			return nil, fmt.Errorf("io.Copy: %w", err)
+		}
+	} else {
+		// Update the header to be multipart/mixed; boundary=$randomBoundary
+		boundary := uuid.NewString()
```

Comment:
I think this can be improved a bit. Please factor out the else block into a new function:
```go
func sendMultipartResponse(w http.ResponseWriter, boundary, dataContentType string, data io.Reader) {

}
```

This helps cut down the size of `respondFromLocalFile` but more importantly the new function can be unit tested with the `httptest` package. Please add a test with the exact example from the MSC (the one with `This media is plain text. Maybe somebody used it as a paste bin.`)